### ISSUE TITLE
Development plans and path to take after speaking with original mycilium developer. (Previously called: Suggestion)

### ISSUE THREAD

**Issue by Hibero** [FuzzyBearBTC/android-peercoinwallet#13]
I think it may be useful to inform the main developer of the mycelium-com wallet that there is a good reward for commits to this code. I think his name is @apetersson.

**Comment by apetersson** [comment]
i just got this notification. please someone from the peercoin intitialtive contact me on skype andreas_petersson - we should have a call, where i am going to explain some of our plans.

**Comment by FuzzyBearBTC** [comment]
hello andreas_petersson many thanks for your comment, I will add you to my skype contacts, very inteested to hear your plans

**Comment by FuzzyBearBTC** [comment]
@apetersson sent you a skype request

**Comment by mark-bl** [comment]
Ok , I can see you are talking with the developer of the app.
I'll stop working on this, leave the funds for the main developer.

**Comment by FuzzyBearBTC** [comment]
skype chat arranged for 18:00 GMT tonight... i';ll update here with info from the meeting

**Comment by FuzzyBearBTC** [comment]
had very productive skype chat with @apetersson  thank you for taking the time to speak with me about your plans and goals for the mycilium wallet, was a pleasure :).

@mark-bl @Hibero To fill you in on our conversation here is a summary.  Let me start by saying Andreas came across as a very knowledgeable and really nice guy so we should definitely look to work with him and the mycilium team on this and the way they see the wallet progressing.

Andreas told me he has some very impressive features that are currently being implemented on the bitcoin mycilium wallet.  I will not give any spoilers as that is for @apetersson to announce but the functionality of these new features we will want in the Peercoin version so we should keep the codebase in line and pull frequently from the main repo.

Once these new features are added the next is to look to have the bitcoin mycilium have multi currency support.  This is where the key lies from our development perspective.  Instead of looking to port the wallet as a stand alone Peercoin android wallet we should look to have a bitcoin and Peercoin wallet running through the same mycilium wallet.  We do not want the Peercoin app to interfere with the bitcoin wallet should a user be running both.  Also this would mean that we develop on our branch here and then we send pull requests to the main repo and Peercoin is available to every user of the bitcoin mycilium wallet (excellent exposure to current bitcoin users it goes without saying).

The way that the mycilium wallet works as well is there is this repo UI code for your mobile app... and there is other code that runs on a clucter of mycilium servers that the mobile apps interigate to get the data.  This code is not public so would either need to be written from scratch by ourselves, or an easier solution would be to work with the mycilium developers to set up the necessary servers for the peercoin code.

Essentially the server runs the bitcoind client and produces an index that the mobile apps can read from... (and write to i guess as well)  Whatever this would need to be made and maintained for Peercoin for this wallet app to work.

As Andreas is interested already in intergrating altcoins into the mycilium wallet his plans over the next few months is to: 
project plan on how to build altcoins, 
start fund raiser for dev of altcoins
Assign bounties for very specific tasks 
mycilium also will take on some tasks and hence claim bounties

We also discussed the possibility of having POS generation in the wallet and simply by the way the mycilium wallet works you would not be able to easily integrate POS minting unless there was a way you could POS mint without the coins live on the blockchain.. essentially needs the intergration of cold wallet offline POS minting to be enabled in the codebase of Peercoin and then the wallet could interact with this and be the online index access to enable POS minting for set addresses.  So we are best to wait for Sunny King on this one... but I would see POS minting as a Phase 2 task.. Phase 1 being to get an android wallet for peercoin.

timescale of next actions from andreas is aimed to be in 2 weeks a draft a plan for poject intergration of altcoins.

In conclusion I am going to put all the work done so far on a seperate branch and revert the master back to the mycilium wallet.. they have just committed and pushed 15,000 lines of code recently!! so big changes / additions etc that we need to pull in.

We should look to start to see how the structure of the code should be for adding Peercoin as a wallet within the same bitcoin wallet so users could interact with both through the same platform.

Fuzzybear

**Comment by Hibero** [comment]
@FuzzyBearBTC I think we should definitely try to keep this wallet in mycilium hands. I think they would have the resources to run it while we do not. (At least that I am aware of.) With this is mind, I would like help out in any way I can. Either in the form of donations for bounties or development work. 

I think this is a project that would be better suited for bounties and not being mainly funded by peer4commit. The tip for commit scheme does have it's advantages but  I feel bounties can bring this to fruition more quickly and more optimally. The faster we can get a working android wallet out, the better.

**Comment by mark-bl** [comment]
This is great to hear, I say stop accepting pull requests and wait for upstream to make a move.
Lets use the funds in order to do work to integrate peercoin with upstream.

**Comment by FuzzyBearBTC** [comment]
http://www.peercointalk.org/index.php?topic=2467.0 minting from cold storage thread

**Comment by Cybnate** [comment]
Someone just donated 4000 Peercoins to this project on peer4commit. Might need to add something under description on peer4commit about the above.

**Comment by FuzzyBearBTC** [comment]
tried to add link to this discussion on the description of this repo, will have to see if it updates on Peer4commit.  I will be updating the Readme.me file to give more up to date version of development plans based off this thread and the skype call I had with Andreas.

**Comment by Cybnate** [comment]
Any updates from the Mycilium project? Development of master seems to be very slow. A bit frustrating as this is a very promising wallet. 

Are we still waiting for those 'new features' to be added? Are there any timelines?
Asking all the questions as I think we should update the donators. Maybe open a thread for each project with its status, progress and what is needed or needs to be done?
Happy to do this, but I do need some input, FB.

**Comment by mark-bl** [comment]
An update will be nice.

I wish I could do more to help, but between University and work , I don't have much free time.

**Comment by FuzzyBearBTC** [comment]
@apetersson sorry I have been very busy of late and not had a chance to follow up with you on this project development, I was wondering if you had an outline for the next development phases in adding altcoins / Peercoin?

**Comment by apetersson** [comment]
i can assure you our development is not so slow. we are very busy but most is currently not ready for push or on the "testnet" branch.

i did promise to give you an outline of our altcoin plans, which is under development. the fulfilment of this promise is still outstanding and will follow soon (tm) please give me a few more days. until we have launched our next major feature (local trading)

**Comment by rhnardini** [comment]
@apetersson Hi Andreas, Great thing if you add altcoins as Feathercoin. Thank you very much for support!

**Comment by Cybnate** [comment]
Hi guys, any updates as of late. Has been a while since the last update. Is there anything we need specifically before work can start, or has it already started on a different repo?

**Comment by Hibero** [comment]
@FuzzyBearBTC Hey, any new developments in this area?

**Comment by FuzzyBearBTC** [comment]
@apetersson would be good to have a catch up to see where development is for adding altcoins, I am very happy to do some work just need to outline the dev process
